src/rapids_dependency_file_generator/rapids_dependency_file_validator.py

```diff
@@ -0,0 +1,155 @@
+                                                "packages": {
+                                                    "type": ["null", "array", "string"],
+                                                    "if": {"type": "array"},
+                                                    "then": {
+                                                        "items": {
+                                                            "type": ["string", "object"]
+                                                        },
+                                                        "if": {"type": "object"},
+                                                        "then": {
+                                                            "patternProperties": {
+                                                                ".*": {
+                                                                    "type": [
+                                                                        "array",
+                                                                        "string",
+                                                                    ],
+                                                                    "if": {
+                                                                        "type": "array"
+                                                                    },
+                                                                    "then": {
+                                                                        "items": {
+                                                                            "type": "string"
+                                                                        }
+                                                                    },
+                                                                }
+                                                            }
+                                                        },
+                                                    },
+                                                },
+                                            },
+                                        },
+                                    },
+                                },
+                            },
+                        },
+                    },
+                }
+            },
+        },
+    },
+    "required": ["files", "dependencies"],
+}
+
+
+def validate_dependencies(dependencies):
+    """Valid a dictionary against the dependencies.yaml spec.
+
+    Parameters
+    ----------
+    dependencies : dict
+        The parsed dependencies.yaml file.
+
+    Raises
+    ------
+    jsonschema.exceptions.ValidationError
+        If the dependencies do not conform to the schema
+    """
+    validator = jsonschema.Draft7Validator(SCHEMA)
+    if not validator.is_valid(dependencies):
+        for i, error in enumerate(validator.iter_errors(dependencies), start=1):
```

Comment:
It looks like [`iter_errors`](https://python-jsonschema.readthedocs.io/en/stable/api/jsonschema/protocols/#jsonschema.protocols.Validator.iter_errors) is deprecated. I would need to investigate whether there is an alternative way to achieve this.

Reply:
We clarified this offline: `iter_errors` itself is not deprecated, only calling it with an extra schema is. The usage here is still valid.